client/apiclient.go

```diff
@@ -229,3 +125,113 @@ func (c *APIClient) request(method string, path string, data []byte) (interface{
 
 	return iface, err
 }
+
```

Comment:
Everything below here is just code that got moved from above, but there were no "new" changes.